static/src/javascripts/projects/commercial/modules/messenger.ts

```diff
@@ -0,0 +1,382 @@
+import reportError from '../../../lib/report-error';
+import { postMessage } from './messenger/post-message';
+
+/**
+ * A message that is sent from an iframe following a standard format
+ *
+ * TODO Is this format formally defined somewhere?
+ */
+type StandardMessage<T = unknown> = {
+	id: string;
+	type: string;
+	iframeId?: string | null | undefined;
+	slotId?: string | null | undefined;
```

Comment:
When converting from Flow, you get a lot of this, but really it’s just:

```suggestion
	iframeId?: string;
	slotId?: string;
```

Reply:
Good point, I started by making the types as permissive as possible and then never removed these, but can have another look.